modules/git/repo_compare.go

```diff
@@ -263,3 +268,40 @@ func (repo *Repository) GetDiffFromMergeBase(base, head string, w io.Writer) err
 	}
 	return err
 }
+
+// ReadPatch will check if a diff patch exists and return stats
+func (repo *Repository) ReadPatch(prID int64) (commitSHA string) {
+	// Migrated repositories download patches to "pulls" location
+	patchFile := fmt.Sprintf("pulls/%d.patch", prID)
+	loadPatch, err := ioutil.ReadFile(filepath.Join(repo.Path, patchFile))
```

Comment:
How about a very large patch? i.e. 1GB?

Reply:
Should I stat and/or read only like the 10 first lines ?

Reply:
I think the commit head is on the first line?

Reply:
Yes, but I noticed Gitlab tends to add lines in end of file... 
I was wondering maybe this behavior could change, thus taking the first 10 lines would be enough. Otherwise we just take 1st line, I can do this